### PR TITLE
Implement SA1603

### LINKED ISSUE #123: SA1603: DocumentationMustContainValidXml
http://www.stylecop.com/docs/SA1603.html

### PR DESCRIPTION
**Pull request by pdelvo**
A code fix would probably in most cases fix it incorrectly so I don't think we should implement one.

Fixes #123

### CODE REVIEW

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1603DocumentationMustContainValidXml.cs [hunk 1]
```diff
@@ -45,7 +50,103 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.EndTag.GetLocation(), "Xml element end tag is missing a '</'."));
+                }
+
+                if (xmlElementSyntax.EndTag.GreaterThanToken.IsMissing)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.EndTag.GetLocation(), "Xml element end tag is missing a '>'."));
+                }
+                if (xmlElementSyntax.StartTag.Name.ToString() != xmlElementSyntax.EndTag.Name.ToString())
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.StartTag.GetLocation(), new[] { xmlElementSyntax.EndTag.GetLocation() }, $"The '{xmlElementSyntax.StartTag.Name}' start tag does not match the end tag of '{xmlElementSyntax.EndTag.Name}'."));
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1603DocumentationMustContainValidXml.cs]: For this particular error, I would report the warning on the end tag with the following message:

> The closing XML tag '{xmlElementSyntax.EndTag.Name}' is mismatched.

File: StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1603DocumentationMustContainValidXml.cs [hunk 2]
```diff
@@ -45,7 +50,103 @@ public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
...
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.StartTag.GetLocation(), "Xml element start tag is missing a '<'."));
+                }
+
+                if (xmlElementSyntax.StartTag.GreaterThanToken.IsMissing)
+                {
+                    context.ReportDiagnostic(Diagnostic.Create(Descriptor, xmlElementSyntax.StartTag.GetLocation(), "Xml element start tag is missing a '>'."));
+                }
+
+                if (xmlElementSyntax.EndTag.LessThanSlashToken.IsMissing)
```
Comment (sharwell) [inline StyleCop.Analyzers/StyleCop.Analyzers/DocumentationRules/SA1603DocumentationMustContainValidXml.cs]: I believe the end tag can be missing altogether:

```csharp
/// <summary>Foo
void Foo() { }
```

If that is the case, I would report the warning on the start tag with the following message:

> The XML tag '{xmlElementSyntax.StartTag.Name}' is not closed.

**Comment by pdelvo** [comment]
I implemented that

**Comment by sharwell** [comment]
I made a few style tweaks during the merge, and also enabled this diagnostic by default. :+1:

**Pull request merged**